boshio/boshio.go

```diff
@@ -194,7 +196,10 @@ func (c Client) retryableRequest(stemcellURL string, byteRange string) ([]byte,
 	byteRangeHeader := fmt.Sprintf("bytes=%s", byteRange)
 	req.Header.Add("Range", byteRangeHeader)
 
-Retry:
+	return c.retry(req)
+}
+
+func (c Client) retry(req *http.Request) ([]byte, error) {
```

Comment:
Lets do the `for` if we can put this all back into one loop and it reads cleaner.